tests/jerry/unusual-javascript-constructs.js

```diff
@@ -0,0 +1,32 @@
+// Copyright JS Foundation and other contributors, http://js.foundation
```

Comment:
"javascript" in the file name doesn't carry too much information (considering that this is the test suite of a javascript engine). I think tests/jerry/unusual.js would be enough.